### PR TITLE
implement ComputationError struct, so applications can index into error metadata

### PR DESCRIPTION
**Pull request by rma-stripe**
Based on the signalfx-python implementation[0][1], which always exposes a code and sometimes exposes a message/error type (if they are available).

[0] https://github.com/signalfx/signalfx-python/blob/cfbda55989bc0e7ff48be82f2510f378d97d30c6/signalfx/signalflow/ws.py#L311-L314
[1] https://github.com/signalfx/signalfx-python/blob/cfbda55989bc0e7ff48be82f2510f378d97d30c6/signalfx/signalflow/errors.py#L28-L35

### CODE REVIEW

File: signalflow/computation_test.go [hunk 1]
```diff
@@ -11,15 +12,25 @@ import (
...
+				continue
+			}
+			return m, nil
+		case <-ctx.Done():
+			err := comp.Err()
+			if err != nil {
+				return nil, err
+			}
+
+			t.Fatal("unepxeted empty error")
```
Comment (cory-signalfx) [inline signalflow/computation_test.go]: Typo here!

**Comment by cory-signalfx** [comment]
Hey @keitwb, thoughts on this?

File: signalflow/computation_test.go (hunk 1 above)
Comment (rma-stripe) [reply signalflow/computation_test.go]: Thanks, resolved.

File: signalflow/computation.go [hunk 2]
```diff
@@ -40,6 +40,24 @@ type Computation struct {
 	MetadataTimeout time.Duration
 }
 
+// ComputationError exposes the underlying metadata of a computation error
+type ComputationError struct {
+	Code      interface{}
```
Comment (keitwb) [inline signalflow/computation.go]: I have confirmed by looking at the backend code that `Code` will always be an `int`, `Message` will always be a `string`, and `ErrorType` will always be a `string`.  I think it is pretty safe using those types even though it isn't officially documented.  Would also make it easier to inspect without having to do a bunch of type assertions/switches in Go.

File: signalflow/computation.go [hunk 3]
```diff
@@ -183,7 +201,13 @@ func (c *Computation) processMessage(m messages.Message) {
 			c.maxDelayMS = pointer.Int(v.MessageBlock.Contents.(messages.JobInitialMaxDelayContents).MaxDelayMS())
 		}
 	case *messages.ErrorMessage:
-		c.lastError = fmt.Errorf("error executing SignalFlow: %v", v.RawData())
+		rawData := v.RawData()
+		computationError := ComputationError{
+			Code:      rawData["error"],
```
Comment (keitwb) [inline signalflow/computation.go]: I have confirmed by looking at the backend code that `error` will always be an `int`, `message` will always be a `string`, and `errorType` will always be a `string`. I think it is pretty safe using those types even though it isn't officially documented. Would also make it easier to inspect without having to do a bunch of type assertions/switches in Go.

**Comment by rma-stripe** [comment]
Thanks @keitwb, I pushed another commit doing exactly that. Are we ok to merge?

**Comment by cory-signalfx** [comment]
Thanks, @rma-stripe! Tag incoming, but got one other one to sneak in.

**Pull request merged**